core/src/main/java/org/apache/calcite/runtime/JsonFunctions.java

```diff
@@ -644,6 +644,35 @@ public static Integer jsonStorageSize(JsonValueContext input) {
     }
   }
 
+  public static String jsonExtract(String input, String... pathSpecs) {
+    return jsonExtract(jsonValueExpression(input), pathSpecs);
+  }
+
+  public static String jsonExtract(JsonValueContext input, String... pathSpecs) {
+    try {
+      DocumentContext ctx = JsonPath.parse(input.obj,
+          Configuration
+              .builder()
+              .options(Option.SUPPRESS_EXCEPTIONS)
+              .jsonProvider(JSON_PATH_JSON_PROVIDER)
+              .mappingProvider(JSON_PATH_MAPPING_PROVIDER)
+              .build());
+      List<Object> arrayList = new ArrayList<Object>();
+      Object value;
+      for (String pathSpec : pathSpecs) {
+        value = ctx.read(pathSpec);
+        if ((pathSpec != null) && (value != null)) {
+          arrayList.add(value);
```

Comment:
It seems parentheses is not necessary in` (pathSpec != null) && (value != null)`.

Reply:
well,let me change it.